src/Components/ColorPicker/ColorPicker.tsx

```diff
@@ -27,58 +32,105 @@ import {validateHexCode} from '../../Utils/index'
 // Styles
 import './ColorPicker.scss'
 
-interface Props extends StandardClassProps {
+interface ColorPickerProps extends StandardFunctionProps {
   /** currently selected color */
   color: string
   /** Function to be called on color select */
   onChange: (color: string, status?: ComponentStatus) => void
   /** Array of colors to be displayed in color picker */
-  colors: Color[]
+  colors?: Color[]
   /** Prevent focus from leaving the input */
-  maintainInputFocus: boolean
+  maintainInputFocus?: boolean
   /** How many color swatches to render in each row */
-  swatchesPerRow: number
+  swatchesPerRow?: number
 }
 
-interface State {
-  errorMessage: string | null
-}
-
-export class ColorPicker extends Component<Props, State> {
-  public static readonly displayName = 'ColorPicker'
+export type ColorPickerRef = HTMLDivElement
 
-  public static defaultProps = {
-    colors: influxColors,
-    maintainInputFocus: false,
-    swatchesPerRow: 10,
-    testID: 'color-picker',
-  }
+export const ColorPicker = forwardRef<ColorPickerSwatchRef, ColorPickerProps>(
+  (
+    {
+      id,
+      style,
+      color,
+      onChange,
+      className,
+      swatchesPerRow = 10,
+      colors = influxColors,
+      testID = 'color-picker',
+      maintainInputFocus = false,
+    },
+    ref
+  ) => {
+    const [errorMessage, setErrorMessage] = useState('')
+
+    const colorPickerClass = classnames('cf-color-picker', {
+      [`${className}`]: className,
+    })
+
+    const inputStatus = errorMessage
+      ? ComponentStatus.Error
+      : ComponentStatus.Valid
+
+    const handleSwatchClick = (hex: string): void => {
+      setErrorMessage('')
+      onChange(hex, ComponentStatus.Valid)
+    }
 
-  constructor(props: Props) {
-    super(props)
+    const handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
+      const acceptedChars = [
```

Comment:
This could probs be a constant (for readability)

Reply:
moved